src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexQuickInfoProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.QuickInfo;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions.LanguageServices
+{
+    [ExportEmbeddedLanguageQuickInfoProvider("Regex", new[] { LanguageNames.CSharp }, new[] { "Regex", "Regexp" }), Shared]
```

Comment:
I think this would work for VB as is. Consider adding `VisualBasic` to the language list after you confirmed it.

Reply:
Yep, absolutely should work.

Reply:
(actually it's an interesting question for me why we even have a language filter like this....@cyrusnajmabadi?)

Reply:
It naturally falls out from Roslyn being multi lang.  And the lowest levels this system supports embedding any language in anything else. Practically, it will just be embedding within c# and vb. But I didn't want to hardcode that as it offended my delicate sensibilities.